perllib/FixMyStreet/Roles/ContactExtra.pm

```diff
@@ -25,8 +25,15 @@ sub for_bodies {
 }
 
 sub by_categories {
-    my ($rs, $area_id, @contacts) = @_;
-    my %body_ids = map { $_->body_id => 1 } FixMyStreet::DB->resultset('BodyArea')->search({ area_id => $area_id });
+    my ($rs, $contacts, %params) = @_;
+
+    my %body_ids = ();
+    if ( $params{body_id} ) {
+        %body_ids = ( $params{body_id} => 1 );
+    } else {
+        %body_ids = map { $_->body_id => 1 } FixMyStreet::DB->resultset('BodyArea')->search({ area_id => $params{area_id} });
```

Comment:
Yeah, you could drop this, and then the below could be simplified if you knew it was only ever one body. But meh.